shell/common/platform_view.h

```diff
@@ -210,6 +210,36 @@ class PlatformView {
     ///
     virtual void OnPlatformViewMarkTextureFrameAvailable(
         int64_t texture_id) = 0;
+
+    //--------------------------------------------------------------------------
+    /// @brief      Loads the dart shared library into the dart VM. When the
+    ///             dart library is loaded successfully, the dart future
+    ///             returned by the originating loadLibrary() call completes.
+    ///             Each shared library is a loading unit, which consists of
+    ///             deferred libraries that can be compiled split from the
+    ///             base dart library by gen_snapshot.
+    ///
+    /// @param[in]  loading_unit_id  The unique id of the deferred library's
+    ///                              loading unit.
+    ///
+    /// @param[in]  snapshot_data    Dart snapshot data of the loading unit's
+    ///                              shared library.
+    ///
+    /// @param[in]  snapshot_data    Dart snapshot instructions of the loading
+    ///                              unit's shared library.
+    ///
+    virtual void CompleteDartDeferredLibraryLoading(
+        intptr_t loading_unit_id,
+        const uint8_t* snapshot_data,
+        const uint8_t* snapshot_instructions) = 0;
+
+    //--------------------------------------------------------------------------
+    /// @brief      Sets the asset manager of the engine to asset_manager
+    ///
+    /// @param[in]  asset_manager  The asset manager to use.
+    ///
+    virtual void UpdateAssetManager(
```

Comment:
I mentioned this in a TODO somewhere, This "replacement" is temporary. I plan on implementing a system to push a new asset resolver, but the challenge is to track and dispose of the previous APKAssetResolver to prevent a leak. For now, this is more of a stand-in implementation. I expect the full implementation to require additional research and/or new features in the asset manager to be able to specifically find and replace a particular asset resolver.

Reply:
ah ok, leave a TODO so future maintainers don't pile onto this

Reply:
There is a TODO in one of the other versions of this, I'll add it here too.